<launch>
  <env name="ROSCONSOLE_FORMAT" value="[${severity}] [${node}]: ${message}"/>
  <rosparam command="delete" param="/"/>

  <!--geometric patternを選択-->
  <arg name="gp" default="line" doc="{line, circle, disk}"/>

  <rosparam command="load" file="$(find coverage_control)/config/base.yaml"/>
  <rosparam command="load" file="$(find coverage_control)/config/geometric.yaml"/>
  <param name="geometric_pattern" value="$(arg gp)"/>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find coverage_control)/rviz/geometric.rviz"/>

  <node name="pose_collector" pkg="coverage_control" type="pose_collector.py"/>

  <include file="$(find coverage_control)/launch/agent.launch">
    <arg name="id" value="0"/>
  </include>
  <include file="$(find coverage_control)/launch/agent.launch">
    <arg name="id" value="1"/>
  </include>
  <include file="$(find coverage_control)/launch/agent.launch">
    <arg name="id" value="2"/>
  </include>
  <include file="$(find coverage_control)/launch/agent.launch">
    <arg name="id" value="3"/>
  </include>
  <include file="$(find coverage_control)/launch/agent.launch">
    <arg name="id" value="4"/>
  </include>
  <include file="$(find coverage_control)/launch/agent.launch">
    <arg name="id" value="5"/>
  </include>
  <include file="$(find coverage_control)/launch/agent.launch">
    <arg name="id" value="6"/>
  </include>
  <include file="$(find coverage_control)/launch/agent.launch">
    <arg name="id" value="7"/>
  </include>
  <include file="$(find coverage_control)/launch/agent.launch">
    <arg name="id" value="8"/>
  </include>
  <include file="$(find coverage_control)/launch/agent.launch">
    <arg name="id" value="9"/>
  </include>
  <include file="$(find coverage_control)/launch/agent.launch">
    <arg name="id" value="10"/>
  </include>

  <!--joyは最後に起動-->
  <node name="joy" pkg="joy" type="joy_node"/>
</launch>